addons/godot-firebase/firestore/firestore.gd

```diff
@@ -26,29 +26,34 @@ signal result_query(result)
 ## @arg-types Dictionary
 signal error(error)
 
-## @enum REQUESTS
-enum REQUESTS {
+enum Requests {
     NONE = -1,  ## Firestore is not processing any request.
     LIST,       ## Firestore is processing a [code]list()[/code] request on a collection.
     QUERY       ## Firestore is processing a [code]query()[/code] request on a collection.
-   }
+}
 
-## @doc-ignore
-const _authorization_header : String = "Authorization: Bearer "
+# TODO: Implement cache size limit
+const CACHE_SIZE_UNLIMITED = -1
+
+const _CACHE_EXTENSION : String = ".fscache"
+const _CACHE_RECORD_FILE : String = "RmlyZXN0b3JlIGNhY2hlLXJlY29yZHMu.fscache"
+
+const _AUTHORIZATION_HEADER : String = "Authorization: Bearer "
+
+const _MAX_POOLED_REQUEST_AGE = 30
 
 ## The code indicating the request Firestore is processing.
-## See @[enum FirebaseFirestore.REQUESTS] to get a full list of codes identifiers.
-## @type int
+## See @[enum FirebaseFirestore.Requests] to get a full list of codes identifiers.
+## @enum Requests
 var request : int = -1
 
-var _base_url : String = "https://firestore.googleapis.com/v1/"
-var _extended_url : String = "projects/[PROJECT_ID]/databases/(default)/documents/"
-var _query_suffix : String = ":runQuery"
-
+## Whether cache files can be used and generated.
+## @default true
+var persistence_enabled : bool = true
```

Comment:
Actually, it won't hurt performance. All it does when enabled is read and write cache files. The performance hit I was talking about earlier was when it was offline and it attempts to make a network request. But I changed it now so that it doesn't bother making a request when offline.